app/ide-desktop/lib/dashboard/src/authentication/src/authentication/providers/session.tsx

```diff
@@ -100,6 +100,13 @@ export const SessionProvider = (props: SessionProviderProps) => {
          * https://github.com/aws-amplify/amplify-js/issues/3391#issuecomment-756473970 */
         window.history.replaceState({}, "", MAIN_PAGE_URL);
         refreshSession();
+        /** Typescript tells us we don't need the final condition because this is an exhaustive
+         * match, but we don't want to turn this into an `else` in case we add more event types we
+         * care about in the future. If we did so, then TypeScript wouldn't notify us about missing
+         * cases. */
+        // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
```

Comment:
This explanation makes a LOT of sense. And this means that this lint is broken in our cases. @somebody1234 do you have some thoughts on it? Maybe we should remove this lint ?

Reply:
i think this lint is useful in many other cases - especially when doing `the_type_says_it_is_not_nullish ?? default_value`. there was also one condition with `Enum.value || boolean` that it caught

so i think it's fine to keep this lint, and disable it in the rare case that it is wrong.

Reply:
Ok, lets do it this way.

Reply:
in this specific case though, i think changing to a `switch`-`case` will resolve the problem. this problem seems to usually occur in chained `if`-`else` where the value gradually gets narrowed by each `if ()` until you end up with only one possible value